orchestrator/gravity_utils/src/types/ethereum_events.rs

```diff
@@ -818,6 +759,106 @@ impl LogicCallExecutedEvent {
     }
 }
 
+fn parse_string(data: &[u8], arg_index: usize) -> Result<String, GravityError> {
+    // fetching the string from the first 32 bytes
+    let offset_start = arg_index * 32;
+    let offset_end = offset_start + 32;
+    let offset = Uint256::from_bytes_be(&data[offset_start..offset_end]).unwrap();
+
+    // parse start and end of the string length
+    let len_start_index: usize = offset.to_string().parse().unwrap();
+    let len_end_index = len_start_index + 32;
+
+    let len = Uint256::from_bytes_be(&data[len_start_index..len_end_index]).unwrap();
+    let len: usize = len.to_string().parse().unwrap();
+    if len == 0 {
+        return Ok("".to_string());
+    }
+
+    let start_index = len_end_index;
+    let end_index = start_index + (((len - 1) / 32) + 1) * 32;
```

Comment:
What is happening with this calculation? Stuff like this should be commented.

Reply:
This is ABI spec. But I've added the comment anyway.